openmdao/drivers/pyoptsparse_driver.py

```diff
@@ -712,7 +712,7 @@ def _objfunc(self, dv_dict):
                 func_dict = self.get_objective_values()
                 func_dict.update(self.get_constraint_values(lintype='nonlinear'))
                 # convert func_dict to use promoted names
-                func_dict = model._prom_names_dict(func_dict)
+                # func_dict = model._prom_names_dict(func_dict)
```

Comment:
Why are we sending absolute names to pyoptsparse now?

Reply:
No the keys stored in the internal dicts are now promoted names or aliases.